AK/bits/Math/libm_fallback.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2021, Leon Albrecht <leon2002.la@gmail.com>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "./common.h"
+
+#include <AK/Concepts.h>
+#include <AK/Types.h>
+#include <math.h>
+
+namespace AK {
+
+namespace Division {
+template<FloatingPoint T>
+CONSTEXPR T fmod(T x, T y) { return ::fmod(x, y); }
+template<FloatingPoint T>
+CONSTEXPR T remainder(T x, T y) { return ::remainder(x, y); }
+}
+
+template<FloatingPoint T>
+CONSTEXPR T sqrt(T x) { return ::sqrt(x); }
+
+template<FloatingPoint T>
+CONSTEXPR T cbrt(T x) { return ::cbrt(x); }
+
+template<FloatingPoint T>
+CONSTEXPR T fabs(T x) { return ::fabs(x); }
+
+namespace Trigonometry {
+template<FloatingPoint T>
+CONSTEXPR T hypot(T x, T y) { return ::hypot(x, y); }
+template<FloatingPoint T>
+CONSTEXPR T sin(T angle) { return ::sin(angle); }
+template<FloatingPoint T>
+CONSTEXPR T cos(T angle) { return ::cos(angle); }
+template<FloatingPoint T>
+CONSTEXPR T tan(T angle) { return ::tan(angle); }
+template<FloatingPoint T>
+CONSTEXPR T atan(T x) { return ::atan(x); }
+template<FloatingPoint T>
+CONSTEXPR T asin(T x) { return ::asin(x); }
+template<FloatingPoint T>
+CONSTEXPR T acos(T x) { return ::acos(x); }
+template<FloatingPoint T>
+CONSTEXPR T atan2(T y, T x) { return ::atan2(y, x); }
+}
+
+namespace Exponentials {
+template<FloatingPoint T>
+CONSTEXPR T log(T x) { return ::log(x); }
+template<FloatingPoint T>
+CONSTEXPR T log2(T x) { return ::log2(x); }
+template<FloatingPoint T>
+CONSTEXPR T log10(T x) { return ::log10(x); }
+template<FloatingPoint T>
+CONSTEXPR T exp(T exponent) { return ::exp(exponent); }
+template<FloatingPoint T>
+CONSTEXPR T exp2(T exponent) { return ::exp2(exponent); }
+}
+
+namespace Hyperbolic {
+template<FloatingPoint T>
+CONSTEXPR T sinh(T x) { return ::sinh(x); }
```

Comment:
Are these ever going to be constant expressions? 🤔

We should probably use function overloading to use the right LibM call based on the type of the argument.

Reply:
I'll change it out with cmath then
I thought normal math had that feature, tooo